addons/hr_attendance/report/hr_attendance_report.py

```diff
@@ -39,7 +39,7 @@ def _from(self):
                             at time zone 'utc'
                             at time zone
                                 (SELECT calendar.tz FROM resource_calendar as calendar
-                                INNER JOIN hr_employee as employee ON employee.id = employee_id
+                                INNER JOIN hr_employee as employee ON employee.id = employee_id::integer
```

Comment:
How about we check `WHERE employee_id IS NOT NULL` instead?  What do you think @atp-odoo?

Reply:
I already tried but it was not working. this error generated specially when there is no any record in
hr_attendance table.